protocol/contracts/ecosystem/ShipmentPlanner.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @notice Constraints of how many Beans to send to a given route at the current time.
+ * @param points Weight of this shipment route relative to all routes. Expects precision of 1e18.
+ * @param cap Maximum Beans that can be received by this stream at this time.
+ */
+struct ShipmentPlan {
+    uint256 points;
+    uint256 cap;
+}
+
+interface IBeanstalk {
+    function isFertilizing() external view returns (bool);
+    function totalUnfertilizedBeans() external view returns (uint256);
+
+    function isHarvesting(uint256 fieldId) external view returns (bool);
+    function totalUnharvestable(uint256 fieldId) external view returns (uint256);
+}
+
+uint256  constant BARN_POINTS = 333_333_333_333_333_333;
+uint256  constant FIELD_POINTS = 333_333_333_333_333_333;
+uint256  constant SILO_POINTS = 333_333_333_333_333_333;
+
+/**
+ * @title ShipmentPlanner
+ * @notice Contains getters for retrieving ShipmentPlans for various Beanstalk components.
+ * @dev Lives as a standalone immutable contract. Updating shipment plans requires deploying
+ *      a new instance and updating the ShipmentRoute planContract addresses help in AppStorage.
+ */
+contract ShipmentPlanner {
+    address constant BEANSTALK = 0xC1E088fC1323b20BCBee9bd1B9fC9546db5624C5;
+
+    IBeanstalk beanstalk = IBeanstalk(BEANSTALK);
+
+    /**
+     * @notice Get the current points and cap for Barn shipments.
+     * @dev The Barn cap is the amount of outstanding unfertilized fertilizer.
+     * @dev data param is unused data to configure plan details.
+     */
+    function getBarnPlan(bytes memory) external view returns (ShipmentPlan memory shipmentPlan) {
+        if (!beanstalk.isFertilizing()) return shipmentPlan;
+        return ShipmentPlan({points: BARN_POINTS, cap: beanstalk.totalUnfertilizedBeans()});
+    }
+
+    /**
+     * @notice Get the current points and cap for Field shipments.
+     * @dev The Field cap is the amount of outstanding Pods unharvestable pods.
+     * @param data Encoded uint256 containing the index of the Field to receive the Beans.
+     */
+    function getFieldPlan(
+        bytes memory data
+    ) external view returns (ShipmentPlan memory shipmentPlan) {
+        uint256 fieldId = abi.decode(data, (uint256));
```

Comment:
can we check that fieldId < s.fieldCount here as well?

Reply:
added
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/d65b6bacc7e6d96105a48c759af2f5222b0ccb26#diff-cf3f4898f41eee71e7b37cb1f5e642c532b4efd2ce1ee23de550f167e61ea6b5R61